lib/elixir/lib/string.ex

```diff
@@ -240,37 +240,48 @@ defmodule String do
       iex> String.printable?("abc" <> <<0>>, 2)
       true
 
+      iex> String.printable?("abc" <> <<0>>, 0)
+      true
+
   """
-  @spec printable?(t) :: boolean
-  @spec printable?(t, non_neg_integer | :infinity) :: boolean
-  def printable?(string, counter \\ :infinity)
+  @spec printable?(t, 0) :: true
+  @spec printable?(t, pos_integer | :infinity) :: boolean
+
+  def printable?(string, character_limit \\ :infinity)
+      when is_binary(string) and
+             (character_limit == :infinity or
```

Comment:
Passing integer limit is likely more common case, I guess we should put it first.

Reply:
I thought that calling `String.printable?/1` would be the most common case.
actually I cannot find anybody using the limit arg
https://github.com/search?utf8=%E2%9C%93&q=%22String.printable%22+language%3Aelixir&type=Code

Reply:
Indeed, String.printable?/1` should be even more common case.